### PR TITLE
Add support to specify DBus event loop

### PR DESCRIPTION
**Pull request by testeddoughnut**
This adds the ability to specify the DBus event loop for backends that use DBus (kwallet).

I'm going to be honest, this is my first time playing with DBus, so any feedback/criticism is welcomed =D.

Signed-off-by: M. David Bennett mdavidbennett@syntheticworks.com

### CODE REVIEW

**Comment by testeddoughnut** [comment]
Closing this for now to do some additional work on it.

File: keyring/core.py [hunk 1]
```diff
@@ -181,5 +182,20 @@ def _load_library_extensions():
...
+def set_dbus_mainloop(mainloop):
+    """
+    Set the DBus event loop for keyring backends that use DBus (kwallet for
+    example). An example of using this in your glib app:
+        import keyring
+        from dbus.mainloop.glib import DBusGMainLoop
+        keyring.set_dbus_mainloop(DBusGMainLoop)
+    Other mainloops:
+        dbus.mainloop.qt.DBusQtMainLoop (PyQT4)
+        dbus.mainloop.qt5.DBusQtMainLoop (PyQT5)
```
Comment (mitya57) [inline keyring/core.py:194]: It's <code>dbus.mainloop.**py**qt5</code>.

**Comment by mitya57** [comment]
What is the use case of this? Keyring uses only synchronous methods, doesn't it?

Also, I think applications can call `Loop(set_as_default=True)` themselves before instantiating Keyring, if they need.

**Comment by testeddoughnut** [comment]
Disregard this, I am dumb =D

**Pull request closed without merging**